pkgs/tools/security/semgrep/default.nix

```diff
@@ -15,26 +15,53 @@ let
 in
 buildPythonApplication rec {
   pname = "semgrep";
-  inherit (common) version;
-  src = "${common.src}/cli";
+  inherit (common) src version;
 
-  SEMGREP_CORE_BIN = "${semgrep-core}/bin/semgrep-core";
+  postPatch = (lib.concatStringsSep "\n" (lib.mapAttrsToList (
+    path: submodule: ''
+      # substitute ${path}
+      # remove git submodule placeholder
+      rm -r ${path}
+      # link submodule
+      ln -s ${submodule}/ ${path}
+    ''
+  ) common.submodules)) + ''
+    cd cli
```

Comment:
is there a more nix-ie way to do this? :shrug:

Reply:
I don't think so, but a comment explaining what is going on and why this is needed would be helpful!

Reply:
Does `fetchSubmodules = true` is a viable option ?
https://github.com/NixOS/nixpkgs/blob/master/doc/builders/fetchers.chapter.md#fetchgit-fetchgit